dbt/adapters/athena/connections.py

```diff
@@ -35,6 +38,63 @@
 logger = AdapterLogger("Athena")
 
 
+def paginate_insert_into(func):
```

Comment:
isn't this affecting all the cases were we do INSERT INTO? did you try to test with materialisation is incremental?

Reply:
This is affecting all times - i didn't test that, what do you mean?
as above, anytime an insert into query is run, we run the risk of running into Athena's query length limit - are you suggesting there are situations where insert into's aren't ever going to run that risk?

Reply:
For example when doing an `insert into` in an incremental materialization the query is limited to the sql in the model, that generally is far from reaching the hard limit of athena.
Said so I don't have much more to suggest, please just be sure that you tested your changes running some basic models, specifically test incremental materailazion with append and insert_overwrite, including iceberg format.